Moonscraper Chart Editor/Assets/Scripts/Game/Charts/IO/Midi/MidWriter.cs

```diff
@@ -649,7 +649,25 @@ static SortableBytes[] GetChartSortableBytes(Song song, Song.Instrument instrume
                 {
                     Starpower sp = chartObject as Starpower;
                     if (sp != null)     // Starpower cannot be split up between charts in a midi file
-                        GetStarpowerBytes(sp, out onEvent, out offEvent);
+                    {
+                        // Starpower notes marked as a drum roll are written as 5 notes instead of 1
+                        SortableBytes[] onEvents;
+                        SortableBytes[] offEvents;
+                        GetStarpowerBytes(sp, out onEvents, out offEvents);
+                        if (onEvents != null && offEvents != null && onEvents.Length == offEvents.Length)
+                        {
+                            // An assumption is made here that the MIDI notes are paired by index
```

Comment:
Assumptions aren't good. Would prefer to change the return value of GetStarpowerBytes to a array/list of pair structs containing a single on and off event instead.

Reply:
Probably an obvious answer, but would you prefer tuples, or KeyValuePairs?

Reply:
I'm going with tuples since KeyValuePairs are gonna be overall more verbose to work with, tuple deconstruction is very handy.